src/crystal/system/win32/event_loop_iocp.cr

```diff
@@ -80,21 +92,29 @@ module Crystal::EventLoop
   def self.create_fd_read_event(io : IO::Evented, edge_triggered : Bool = false) : Crystal::Event
     Crystal::Event.new(Fiber.current)
   end
+
+  def self.create_timeout_event(fiber)
+    Crystal::Event.new(fiber, timeout: true)
+  end
 end
 
 struct Crystal::Event
   getter fiber
   getter wake_at
+  getter? timeout
 
-  def initialize(@fiber : Fiber)
-    @wake_at = Time.monotonic
+  def initialize(@fiber : Fiber, @wake_at = Time.monotonic, *, @timeout = false)
   end
 
   # Frees the event
   def free : Nil
     Crystal::EventLoop.dequeue(self)
   end
 
+  def delete
+    free
```

Comment:
And here is another `dequeue`. From where is this called?

Reply:
That's called from `Fiber#cancel_timeout` which is used for the `select` implementation.